// kalshi-typescript/dist/api/live-data-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveDataApi = exports.LiveDataApiFactory = exports.LiveDataApiFp = exports.LiveDataApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * LiveDataApi - axios parameter creator
 */
const LiveDataApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get live data for a specific milestone
         * @summary Get Live Data
         * @param {string} type Type of live data
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getLiveData: (type_1, milestoneId_1, ...args_1) => __awaiter(this, [type_1, milestoneId_1, ...args_1], void 0, function* (type, milestoneId, options = {}) {
            // verify required parameter 'type' is not null or undefined
            (0, common_1.assertParamExists)('getLiveData', 'type', type);
            // verify required parameter 'milestoneId' is not null or undefined
            (0, common_1.assertParamExists)('getLiveData', 'milestoneId', milestoneId);
            const localVarPath = `/live_data/{type}/milestone/{milestone_id}`
                .replace(`{${"type"}}`, encodeURIComponent(String(type)))
                .replace(`{${"milestone_id"}}`, encodeURIComponent(String(milestoneId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Get live data for multiple milestones
         * @summary Get Multiple Live Data
         * @param {Array<string>} milestoneIds Array of milestone IDs
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getLiveDatas: (milestoneIds_1, ...args_1) => __awaiter(this, [milestoneIds_1, ...args_1], void 0, function* (milestoneIds, options = {}) {
            // verify required parameter 'milestoneIds' is not null or undefined
            (0, common_1.assertParamExists)('getLiveDatas', 'milestoneIds', milestoneIds);
            const localVarPath = `/live_data/batch`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (milestoneIds) {
                localVarQueryParameter['milestone_ids'] = milestoneIds;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.LiveDataApiAxiosParamCreator = LiveDataApiAxiosParamCreator;
/**
 * LiveDataApi - functional programming interface
 */
const LiveDataApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.LiveDataApiAxiosParamCreator)(configuration);
    return {
        /**
         * Get live data for a specific milestone
         * @summary Get Live Data
         * @param {string} type Type of live data
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getLiveData(type, milestoneId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getLiveData(type, milestoneId, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Get live data for multiple milestones
         * @summary Get Multiple Live Data
         * @param {Array<string>} milestoneIds Array of milestone IDs
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getLiveDatas(milestoneIds, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getLiveDatas(milestoneIds, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
exports.LiveDataApiFp = LiveDataApiFp;
/**
 * LiveDataApi - factory interface
 */
const LiveDataApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.LiveDataApiFp)(configuration);
    return {
        /**
         * Get live data for a specific milestone
         * @summary Get Live Data
         * @param {string} type Type of live data
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getLiveData(type, milestoneId, options) {
            return localVarFp.getLiveData(type, milestoneId, options).then((request) => request(axios, basePath));
        },
        /**
         * Get live data for multiple milestones
         * @summary Get Multiple Live Data
         * @param {Array<string>} milestoneIds Array of milestone IDs
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getLiveDatas(milestoneIds, options) {
            return localVarFp.getLiveDatas(milestoneIds, options).then((request) => request(axios, basePath));
        },
    };
};
exports.LiveDataApiFactory = LiveDataApiFactory;
/**
 * LiveDataApi - object-oriented interface
 */
class LiveDataApi extends base_1.BaseAPI {
    /**
     * Get live data for a specific milestone
     * @summary Get Live Data
     * @param {string} type Type of live data
     * @param {string} milestoneId Milestone ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getLiveData(type, milestoneId, options) {
        return (0, exports.LiveDataApiFp)(this.configuration).getLiveData(type, milestoneId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get live data for multiple milestones
     * @summary Get Multiple Live Data
     * @param {Array<string>} milestoneIds Array of milestone IDs
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getLiveDatas(milestoneIds, options) {
        return (0, exports.LiveDataApiFp)(this.configuration).getLiveDatas(milestoneIds, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.LiveDataApi = LiveDataApi;
