// kalshi-typescript/dist/api/multivariate-collections-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trading API
 * Complete API for the Kalshi trading platform including all handlers for SDK generation
 *
 * The version of the OpenAPI document: 2.0.0
 * Contact: support@kalshi.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultivariateCollectionsApi = exports.MultivariateCollectionsApiFactory = exports.MultivariateCollectionsApiFp = exports.MultivariateCollectionsApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * MultivariateCollectionsApi - axios parameter creator
 */
const MultivariateCollectionsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get a single multivariate event collection by ticker
         * @summary Get Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollection: (collectionTicker_1, ...args_1) => __awaiter(this, [collectionTicker_1, ...args_1], void 0, function* (collectionTicker, options = {}) {
            // verify required parameter 'collectionTicker' is not null or undefined
            (0, common_1.assertParamExists)('getMultivariateEventCollection', 'collectionTicker', collectionTicker);
            const localVarPath = `/multivariate_event_collections/{collection_ticker}`
                .replace(`{${"collection_ticker"}}`, encodeURIComponent(String(collectionTicker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Get all multivariate event collections
         * @summary Get Multivariate Event Collections
         * @param {string} [status] Filter by multivariate collection status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollections: (status_1, ...args_1) => __awaiter(this, [status_1, ...args_1], void 0, function* (status, options = {}) {
            const localVarPath = `/multivariate_event_collections`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (status !== undefined) {
                localVarQueryParameter['status'] = status;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Lookup a bundle in a multivariate event collection
         * @summary Lookup Multivariate Event Collection Bundle
         * @param {string} collectionTicker Collection ticker
         * @param {LookupBundleRequest} lookupBundleRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupMultivariateEventCollectionBundle: (collectionTicker_1, lookupBundleRequest_1, ...args_1) => __awaiter(this, [collectionTicker_1, lookupBundleRequest_1, ...args_1], void 0, function* (collectionTicker, lookupBundleRequest, options = {}) {
            // verify required parameter 'collectionTicker' is not null or undefined
            (0, common_1.assertParamExists)('lookupMultivariateEventCollectionBundle', 'collectionTicker', collectionTicker);
            // verify required parameter 'lookupBundleRequest' is not null or undefined
            (0, common_1.assertParamExists)('lookupMultivariateEventCollectionBundle', 'lookupBundleRequest', lookupBundleRequest);
            const localVarPath = `/multivariate_event_collections/{collection_ticker}/lookup`
                .replace(`{${"collection_ticker"}}`, encodeURIComponent(String(collectionTicker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(lookupBundleRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.MultivariateCollectionsApiAxiosParamCreator = MultivariateCollectionsApiAxiosParamCreator;
/**
 * MultivariateCollectionsApi - functional programming interface
 */
const MultivariateCollectionsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.MultivariateCollectionsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Get a single multivariate event collection by ticker
         * @summary Get Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollection(collectionTicker, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMultivariateEventCollection(collectionTicker, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Get all multivariate event collections
         * @summary Get Multivariate Event Collections
         * @param {string} [status] Filter by multivariate collection status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollections(status, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMultivariateEventCollections(status, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Lookup a bundle in a multivariate event collection
         * @summary Lookup Multivariate Event Collection Bundle
         * @param {string} collectionTicker Collection ticker
         * @param {LookupBundleRequest} lookupBundleRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupMultivariateEventCollectionBundle(collectionTicker, lookupBundleRequest, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.lookupMultivariateEventCollectionBundle(collectionTicker, lookupBundleRequest, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
exports.MultivariateCollectionsApiFp = MultivariateCollectionsApiFp;
/**
 * MultivariateCollectionsApi - factory interface
 */
const MultivariateCollectionsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.MultivariateCollectionsApiFp)(configuration);
    return {
        /**
         * Get a single multivariate event collection by ticker
         * @summary Get Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollection(collectionTicker, options) {
            return localVarFp.getMultivariateEventCollection(collectionTicker, options).then((request) => request(axios, basePath));
        },
        /**
         * Get all multivariate event collections
         * @summary Get Multivariate Event Collections
         * @param {string} [status] Filter by multivariate collection status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollections(status, options) {
            return localVarFp.getMultivariateEventCollections(status, options).then((request) => request(axios, basePath));
        },
        /**
         * Lookup a bundle in a multivariate event collection
         * @summary Lookup Multivariate Event Collection Bundle
         * @param {string} collectionTicker Collection ticker
         * @param {LookupBundleRequest} lookupBundleRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupMultivariateEventCollectionBundle(collectionTicker, lookupBundleRequest, options) {
            return localVarFp.lookupMultivariateEventCollectionBundle(collectionTicker, lookupBundleRequest, options).then((request) => request(axios, basePath));
        },
    };
};
exports.MultivariateCollectionsApiFactory = MultivariateCollectionsApiFactory;
/**
 * MultivariateCollectionsApi - object-oriented interface
 */
class MultivariateCollectionsApi extends base_1.BaseAPI {
    /**
     * Get a single multivariate event collection by ticker
     * @summary Get Multivariate Event Collection
     * @param {string} collectionTicker Collection ticker
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMultivariateEventCollection(collectionTicker, options) {
        return (0, exports.MultivariateCollectionsApiFp)(this.configuration).getMultivariateEventCollection(collectionTicker, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get all multivariate event collections
     * @summary Get Multivariate Event Collections
     * @param {string} [status] Filter by multivariate collection status
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMultivariateEventCollections(status, options) {
        return (0, exports.MultivariateCollectionsApiFp)(this.configuration).getMultivariateEventCollections(status, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Lookup a bundle in a multivariate event collection
     * @summary Lookup Multivariate Event Collection Bundle
     * @param {string} collectionTicker Collection ticker
     * @param {LookupBundleRequest} lookupBundleRequest
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    lookupMultivariateEventCollectionBundle(collectionTicker, lookupBundleRequest, options) {
        return (0, exports.MultivariateCollectionsApiFp)(this.configuration).lookupMultivariateEventCollectionBundle(collectionTicker, lookupBundleRequest, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.MultivariateCollectionsApi = MultivariateCollectionsApi;
