// kalshi-typescript/dist/api/search-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchApi = exports.SearchApiFactory = exports.SearchApiFp = exports.SearchApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SearchApi - axios parameter creator
 */
const SearchApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
         * @summary Get Filters for Sports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFiltersForSports: (...args_1) => __awaiter(this, [...args_1], void 0, function* (options = {}) {
            const localVarPath = `/search/filters_by_sport`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
         * @summary Get Tags for Series Categories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTagsForSeriesCategories: (...args_1) => __awaiter(this, [...args_1], void 0, function* (options = {}) {
            const localVarPath = `/search/tags_by_categories`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.SearchApiAxiosParamCreator = SearchApiAxiosParamCreator;
/**
 * SearchApi - functional programming interface
 */
const SearchApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.SearchApiAxiosParamCreator)(configuration);
    return {
        /**
         * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
         * @summary Get Filters for Sports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFiltersForSports(options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getFiltersForSports(options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
         * @summary Get Tags for Series Categories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTagsForSeriesCategories(options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getTagsForSeriesCategories(options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
exports.SearchApiFp = SearchApiFp;
/**
 * SearchApi - factory interface
 */
const SearchApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.SearchApiFp)(configuration);
    return {
        /**
         * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
         * @summary Get Filters for Sports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFiltersForSports(options) {
            return localVarFp.getFiltersForSports(options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
         * @summary Get Tags for Series Categories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTagsForSeriesCategories(options) {
            return localVarFp.getTagsForSeriesCategories(options).then((request) => request(axios, basePath));
        },
    };
};
exports.SearchApiFactory = SearchApiFactory;
/**
 * SearchApi - object-oriented interface
 */
class SearchApi extends base_1.BaseAPI {
    /**
     * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
     * @summary Get Filters for Sports
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getFiltersForSports(options) {
        return (0, exports.SearchApiFp)(this.configuration).getFiltersForSports(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
     * @summary Get Tags for Series Categories
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getTagsForSeriesCategories(options) {
        return (0, exports.SearchApiFp)(this.configuration).getTagsForSeriesCategories(options).then((request) => request(this.axios, this.basePath));
    }
}
exports.SearchApi = SearchApi;
