// kalshi-typescript/dist/api/series-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trading API
 * Complete API for the Kalshi trading platform including all handlers for SDK generation
 *
 * The version of the OpenAPI document: 2.0.0
 * Contact: support@kalshi.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeriesApi = exports.SeriesApiFactory = exports.SeriesApiFp = exports.SeriesApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SeriesApi - axios parameter creator
 */
const SeriesApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get all market series
         * @summary Get Series
         * @param {string} [status] Filter by series status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeries: (status_1, ...args_1) => __awaiter(this, [status_1, ...args_1], void 0, function* (status, options = {}) {
            const localVarPath = `/series`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (status !== undefined) {
                localVarQueryParameter['status'] = status;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Get a single series by its ticker
         * @summary Get Series by Ticker
         * @param {string} ticker The series ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeriesByTicker: (ticker_1, ...args_1) => __awaiter(this, [ticker_1, ...args_1], void 0, function* (ticker, options = {}) {
            // verify required parameter 'ticker' is not null or undefined
            (0, common_1.assertParamExists)('getSeriesByTicker', 'ticker', ticker);
            const localVarPath = `/series/{ticker}`
                .replace(`{${"ticker"}}`, encodeURIComponent(String(ticker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.SeriesApiAxiosParamCreator = SeriesApiAxiosParamCreator;
/**
 * SeriesApi - functional programming interface
 */
const SeriesApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.SeriesApiAxiosParamCreator)(configuration);
    return {
        /**
         * Get all market series
         * @summary Get Series
         * @param {string} [status] Filter by series status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeries(status, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getSeries(status, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Get a single series by its ticker
         * @summary Get Series by Ticker
         * @param {string} ticker The series ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeriesByTicker(ticker, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getSeriesByTicker(ticker, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
exports.SeriesApiFp = SeriesApiFp;
/**
 * SeriesApi - factory interface
 */
const SeriesApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.SeriesApiFp)(configuration);
    return {
        /**
         * Get all market series
         * @summary Get Series
         * @param {string} [status] Filter by series status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeries(status, options) {
            return localVarFp.getSeries(status, options).then((request) => request(axios, basePath));
        },
        /**
         * Get a single series by its ticker
         * @summary Get Series by Ticker
         * @param {string} ticker The series ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeriesByTicker(ticker, options) {
            return localVarFp.getSeriesByTicker(ticker, options).then((request) => request(axios, basePath));
        },
    };
};
exports.SeriesApiFactory = SeriesApiFactory;
/**
 * SeriesApi - object-oriented interface
 */
class SeriesApi extends base_1.BaseAPI {
    /**
     * Get all market series
     * @summary Get Series
     * @param {string} [status] Filter by series status
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSeries(status, options) {
        return (0, exports.SeriesApiFp)(this.configuration).getSeries(status, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get a single series by its ticker
     * @summary Get Series by Ticker
     * @param {string} ticker The series ticker
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSeriesByTicker(ticker, options) {
        return (0, exports.SeriesApiFp)(this.configuration).getSeriesByTicker(ticker, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.SeriesApi = SeriesApi;
