// kalshi-typescript/dist/esm/api/portfolio-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// URLSearchParams not necessarily used
// @ts-ignore
import { URL } from 'url';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, setApiKeyToObject, setSearchParams, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * PortfolioApi - axios parameter creator
 */
export const PortfolioApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *  Endpoint for getting the balance and portfolio value of a member. Both values are returned in cents.
         * @summary Get Balance
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getBalance: (...args_1) => __awaiter(this, [...args_1], void 0, function* (options = {}) {
            const localVarPath = `/portfolio/balance`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication kalshiAccessSignature required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-SIGNATURE", configuration);
            // authentication kalshiAccessKey required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-KEY", configuration);
            // authentication kalshiAccessTimestamp required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-TIMESTAMP", configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *  Endpoint for getting all fills for the member. A fill is when a trade you have is matched.
         * @summary Get Fills
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [orderId] Filter by order ID
         * @param {number} [minTs] Filter items after this Unix timestamp
         * @param {number} [maxTs] Filter items before this Unix timestamp
         * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
         * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFills: (ticker_1, orderId_1, minTs_1, maxTs_1, limit_1, cursor_1, ...args_1) => __awaiter(this, [ticker_1, orderId_1, minTs_1, maxTs_1, limit_1, cursor_1, ...args_1], void 0, function* (ticker, orderId, minTs, maxTs, limit, cursor, options = {}) {
            const localVarPath = `/portfolio/fills`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication kalshiAccessSignature required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-SIGNATURE", configuration);
            // authentication kalshiAccessKey required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-KEY", configuration);
            // authentication kalshiAccessTimestamp required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-TIMESTAMP", configuration);
            if (ticker !== undefined) {
                localVarQueryParameter['ticker'] = ticker;
            }
            if (orderId !== undefined) {
                localVarQueryParameter['order_id'] = orderId;
            }
            if (minTs !== undefined) {
                localVarQueryParameter['min_ts'] = minTs;
            }
            if (maxTs !== undefined) {
                localVarQueryParameter['max_ts'] = maxTs;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (cursor !== undefined) {
                localVarQueryParameter['cursor'] = cursor;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *  Endpoint for getting the total value, in cents, of resting orders. This endpoint is only intended for use by FCM members (rare). Note: If you\'re uncertain about this endpoint, it likely does not apply to you.
         * @summary Get Total Resting Order Value
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPortfolioRestingOrderTotalValue: (...args_1) => __awaiter(this, [...args_1], void 0, function* (options = {}) {
            const localVarPath = `/portfolio/summary/total_resting_order_value`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication kalshiAccessSignature required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-SIGNATURE", configuration);
            // authentication kalshiAccessKey required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-KEY", configuration);
            // authentication kalshiAccessTimestamp required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-TIMESTAMP", configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted: position, total_traded
         * @summary Get Positions
         * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. Use the value returned from the previous response to get the next page.
         * @param {number} [limit] Parameter to specify the number of results per page. Defaults to 100.
         * @param {string} [countFilter] Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted - position, total_traded
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPositions: (cursor_1, limit_1, countFilter_1, ticker_1, eventTicker_1, ...args_1) => __awaiter(this, [cursor_1, limit_1, countFilter_1, ticker_1, eventTicker_1, ...args_1], void 0, function* (cursor, limit, countFilter, ticker, eventTicker, options = {}) {
            const localVarPath = `/portfolio/positions`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication kalshiAccessSignature required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-SIGNATURE", configuration);
            // authentication kalshiAccessKey required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-KEY", configuration);
            // authentication kalshiAccessTimestamp required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-TIMESTAMP", configuration);
            if (cursor !== undefined) {
                localVarQueryParameter['cursor'] = cursor;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (countFilter !== undefined) {
                localVarQueryParameter['count_filter'] = countFilter;
            }
            if (ticker !== undefined) {
                localVarQueryParameter['ticker'] = ticker;
            }
            if (eventTicker !== undefined) {
                localVarQueryParameter['event_ticker'] = eventTicker;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *  Endpoint for getting the member\'s settlements historical track.
         * @summary Get Settlements
         * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
         * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
         * @param {number} [minTs] Filter items after this Unix timestamp
         * @param {number} [maxTs] Filter items before this Unix timestamp
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSettlements: (limit_1, cursor_1, ticker_1, eventTicker_1, minTs_1, maxTs_1, ...args_1) => __awaiter(this, [limit_1, cursor_1, ticker_1, eventTicker_1, minTs_1, maxTs_1, ...args_1], void 0, function* (limit, cursor, ticker, eventTicker, minTs, maxTs, options = {}) {
            const localVarPath = `/portfolio/settlements`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication kalshiAccessSignature required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-SIGNATURE", configuration);
            // authentication kalshiAccessKey required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-KEY", configuration);
            // authentication kalshiAccessTimestamp required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-TIMESTAMP", configuration);
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (cursor !== undefined) {
                localVarQueryParameter['cursor'] = cursor;
            }
            if (ticker !== undefined) {
                localVarQueryParameter['ticker'] = ticker;
            }
            if (eventTicker !== undefined) {
                localVarQueryParameter['event_ticker'] = eventTicker;
            }
            if (minTs !== undefined) {
                localVarQueryParameter['min_ts'] = minTs;
            }
            if (maxTs !== undefined) {
                localVarQueryParameter['max_ts'] = maxTs;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * PortfolioApi - functional programming interface
 */
export const PortfolioApiFp = function (configuration) {
    const localVarAxiosParamCreator = PortfolioApiAxiosParamCreator(configuration);
    return {
        /**
         *  Endpoint for getting the balance and portfolio value of a member. Both values are returned in cents.
         * @summary Get Balance
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getBalance(options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getBalance(options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         *  Endpoint for getting all fills for the member. A fill is when a trade you have is matched.
         * @summary Get Fills
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [orderId] Filter by order ID
         * @param {number} [minTs] Filter items after this Unix timestamp
         * @param {number} [maxTs] Filter items before this Unix timestamp
         * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
         * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFills(ticker, orderId, minTs, maxTs, limit, cursor, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getFills(ticker, orderId, minTs, maxTs, limit, cursor, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         *  Endpoint for getting the total value, in cents, of resting orders. This endpoint is only intended for use by FCM members (rare). Note: If you\'re uncertain about this endpoint, it likely does not apply to you.
         * @summary Get Total Resting Order Value
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPortfolioRestingOrderTotalValue(options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPortfolioRestingOrderTotalValue(options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted: position, total_traded
         * @summary Get Positions
         * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. Use the value returned from the previous response to get the next page.
         * @param {number} [limit] Parameter to specify the number of results per page. Defaults to 100.
         * @param {string} [countFilter] Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted - position, total_traded
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPositions(cursor, limit, countFilter, ticker, eventTicker, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPositions(cursor, limit, countFilter, ticker, eventTicker, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         *  Endpoint for getting the member\'s settlements historical track.
         * @summary Get Settlements
         * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
         * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
         * @param {number} [minTs] Filter items after this Unix timestamp
         * @param {number} [maxTs] Filter items before this Unix timestamp
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSettlements(limit, cursor, ticker, eventTicker, minTs, maxTs, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getSettlements(limit, cursor, ticker, eventTicker, minTs, maxTs, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
/**
 * PortfolioApi - factory interface
 */
export const PortfolioApiFactory = function (configuration, basePath, axios) {
    const localVarFp = PortfolioApiFp(configuration);
    return {
        /**
         *  Endpoint for getting the balance and portfolio value of a member. Both values are returned in cents.
         * @summary Get Balance
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getBalance(options) {
            return localVarFp.getBalance(options).then((request) => request(axios, basePath));
        },
        /**
         *  Endpoint for getting all fills for the member. A fill is when a trade you have is matched.
         * @summary Get Fills
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [orderId] Filter by order ID
         * @param {number} [minTs] Filter items after this Unix timestamp
         * @param {number} [maxTs] Filter items before this Unix timestamp
         * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
         * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFills(ticker, orderId, minTs, maxTs, limit, cursor, options) {
            return localVarFp.getFills(ticker, orderId, minTs, maxTs, limit, cursor, options).then((request) => request(axios, basePath));
        },
        /**
         *  Endpoint for getting the total value, in cents, of resting orders. This endpoint is only intended for use by FCM members (rare). Note: If you\'re uncertain about this endpoint, it likely does not apply to you.
         * @summary Get Total Resting Order Value
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPortfolioRestingOrderTotalValue(options) {
            return localVarFp.getPortfolioRestingOrderTotalValue(options).then((request) => request(axios, basePath));
        },
        /**
         * Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted: position, total_traded
         * @summary Get Positions
         * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. Use the value returned from the previous response to get the next page.
         * @param {number} [limit] Parameter to specify the number of results per page. Defaults to 100.
         * @param {string} [countFilter] Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted - position, total_traded
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPositions(cursor, limit, countFilter, ticker, eventTicker, options) {
            return localVarFp.getPositions(cursor, limit, countFilter, ticker, eventTicker, options).then((request) => request(axios, basePath));
        },
        /**
         *  Endpoint for getting the member\'s settlements historical track.
         * @summary Get Settlements
         * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
         * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
         * @param {string} [ticker] Filter by market ticker
         * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
         * @param {number} [minTs] Filter items after this Unix timestamp
         * @param {number} [maxTs] Filter items before this Unix timestamp
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSettlements(limit, cursor, ticker, eventTicker, minTs, maxTs, options) {
            return localVarFp.getSettlements(limit, cursor, ticker, eventTicker, minTs, maxTs, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * PortfolioApi - object-oriented interface
 */
export class PortfolioApi extends BaseAPI {
    /**
     *  Endpoint for getting the balance and portfolio value of a member. Both values are returned in cents.
     * @summary Get Balance
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getBalance(options) {
        return PortfolioApiFp(this.configuration).getBalance(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *  Endpoint for getting all fills for the member. A fill is when a trade you have is matched.
     * @summary Get Fills
     * @param {string} [ticker] Filter by market ticker
     * @param {string} [orderId] Filter by order ID
     * @param {number} [minTs] Filter items after this Unix timestamp
     * @param {number} [maxTs] Filter items before this Unix timestamp
     * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
     * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getFills(ticker, orderId, minTs, maxTs, limit, cursor, options) {
        return PortfolioApiFp(this.configuration).getFills(ticker, orderId, minTs, maxTs, limit, cursor, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *  Endpoint for getting the total value, in cents, of resting orders. This endpoint is only intended for use by FCM members (rare). Note: If you\'re uncertain about this endpoint, it likely does not apply to you.
     * @summary Get Total Resting Order Value
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getPortfolioRestingOrderTotalValue(options) {
        return PortfolioApiFp(this.configuration).getPortfolioRestingOrderTotalValue(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted: position, total_traded
     * @summary Get Positions
     * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. Use the value returned from the previous response to get the next page.
     * @param {number} [limit] Parameter to specify the number of results per page. Defaults to 100.
     * @param {string} [countFilter] Restricts the positions to those with any of following fields with non-zero values, as a comma separated list. The following values are accepted - position, total_traded
     * @param {string} [ticker] Filter by market ticker
     * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getPositions(cursor, limit, countFilter, ticker, eventTicker, options) {
        return PortfolioApiFp(this.configuration).getPositions(cursor, limit, countFilter, ticker, eventTicker, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *  Endpoint for getting the member\'s settlements historical track.
     * @summary Get Settlements
     * @param {number} [limit] Number of results per page. Defaults to 100. Maximum value is 200.
     * @param {string} [cursor] Pagination cursor. Use the cursor value returned from the previous response to get the next page of results. Leave empty for the first page.
     * @param {string} [ticker] Filter by market ticker
     * @param {string} [eventTicker] Event ticker of desired positions. Multiple event tickers can be provided as a comma-separated list (maximum 10).
     * @param {number} [minTs] Filter items after this Unix timestamp
     * @param {number} [maxTs] Filter items before this Unix timestamp
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSettlements(limit, cursor, ticker, eventTicker, minTs, maxTs, options) {
        return PortfolioApiFp(this.configuration).getSettlements(limit, cursor, ticker, eventTicker, minTs, maxTs, options).then((request) => request(this.axios, this.basePath));
    }
}
