// kalshi-typescript/dist/esm/api/series-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trading API
 * Complete API for the Kalshi trading platform including all handlers for SDK generation
 *
 * The version of the OpenAPI document: 2.0.0
 * Contact: support@kalshi.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// URLSearchParams not necessarily used
// @ts-ignore
import { URL } from 'url';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setBearerAuthToObject, setSearchParams, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * SeriesApi - axios parameter creator
 */
export const SeriesApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get all market series
         * @summary Get Series
         * @param {string} [status] Filter by series status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeries: (status_1, ...args_1) => __awaiter(this, [status_1, ...args_1], void 0, function* (status, options = {}) {
            const localVarPath = `/series`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (status !== undefined) {
                localVarQueryParameter['status'] = status;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Get a single series by its ticker
         * @summary Get Series by Ticker
         * @param {string} ticker The series ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeriesByTicker: (ticker_1, ...args_1) => __awaiter(this, [ticker_1, ...args_1], void 0, function* (ticker, options = {}) {
            // verify required parameter 'ticker' is not null or undefined
            assertParamExists('getSeriesByTicker', 'ticker', ticker);
            const localVarPath = `/series/{ticker}`
                .replace(`{${"ticker"}}`, encodeURIComponent(String(ticker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * SeriesApi - functional programming interface
 */
export const SeriesApiFp = function (configuration) {
    const localVarAxiosParamCreator = SeriesApiAxiosParamCreator(configuration);
    return {
        /**
         * Get all market series
         * @summary Get Series
         * @param {string} [status] Filter by series status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeries(status, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getSeries(status, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Get a single series by its ticker
         * @summary Get Series by Ticker
         * @param {string} ticker The series ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeriesByTicker(ticker, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getSeriesByTicker(ticker, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
/**
 * SeriesApi - factory interface
 */
export const SeriesApiFactory = function (configuration, basePath, axios) {
    const localVarFp = SeriesApiFp(configuration);
    return {
        /**
         * Get all market series
         * @summary Get Series
         * @param {string} [status] Filter by series status
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeries(status, options) {
            return localVarFp.getSeries(status, options).then((request) => request(axios, basePath));
        },
        /**
         * Get a single series by its ticker
         * @summary Get Series by Ticker
         * @param {string} ticker The series ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getSeriesByTicker(ticker, options) {
            return localVarFp.getSeriesByTicker(ticker, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * SeriesApi - object-oriented interface
 */
export class SeriesApi extends BaseAPI {
    /**
     * Get all market series
     * @summary Get Series
     * @param {string} [status] Filter by series status
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSeries(status, options) {
        return SeriesApiFp(this.configuration).getSeries(status, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get a single series by its ticker
     * @summary Get Series by Ticker
     * @param {string} ticker The series ticker
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getSeriesByTicker(ticker, options) {
        return SeriesApiFp(this.configuration).getSeriesByTicker(ticker, options).then((request) => request(this.axios, this.basePath));
    }
}
