// kalshi-typescript/dist/api/milestones-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trading API
 * Complete API for the Kalshi trading platform including all handlers for SDK generation
 *
 * The version of the OpenAPI document: 2.0.0
 * Contact: support@kalshi.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MilestonesApi = exports.MilestonesApiFactory = exports.MilestonesApiFp = exports.MilestonesApiAxiosParamCreator = void 0;
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * MilestonesApi - axios parameter creator
 */
const MilestonesApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get a single milestone by ID
         * @summary Get Milestone
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestone: (milestoneId_1, ...args_1) => __awaiter(this, [milestoneId_1, ...args_1], void 0, function* (milestoneId, options = {}) {
            // verify required parameter 'milestoneId' is not null or undefined
            (0, common_1.assertParamExists)('getMilestone', 'milestoneId', milestoneId);
            const localVarPath = `/milestones/{milestone_id}`
                .replace(`{${"milestone_id"}}`, encodeURIComponent(String(milestoneId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Get all milestones
         * @summary Get Milestones
         * @param {string} [status] Filter by milestone status
         * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestones: (status_1, limit_1, ...args_1) => __awaiter(this, [status_1, limit_1, ...args_1], void 0, function* (status, limit, options = {}) {
            const localVarPath = `/milestones`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            if (status !== undefined) {
                localVarQueryParameter['status'] = status;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.MilestonesApiAxiosParamCreator = MilestonesApiAxiosParamCreator;
/**
 * MilestonesApi - functional programming interface
 */
const MilestonesApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.MilestonesApiAxiosParamCreator)(configuration);
    return {
        /**
         * Get a single milestone by ID
         * @summary Get Milestone
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestone(milestoneId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMilestone(milestoneId, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Get all milestones
         * @summary Get Milestones
         * @param {string} [status] Filter by milestone status
         * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestones(status, limit, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMilestones(status, limit, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
exports.MilestonesApiFp = MilestonesApiFp;
/**
 * MilestonesApi - factory interface
 */
const MilestonesApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.MilestonesApiFp)(configuration);
    return {
        /**
         * Get a single milestone by ID
         * @summary Get Milestone
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestone(milestoneId, options) {
            return localVarFp.getMilestone(milestoneId, options).then((request) => request(axios, basePath));
        },
        /**
         * Get all milestones
         * @summary Get Milestones
         * @param {string} [status] Filter by milestone status
         * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestones(status, limit, options) {
            return localVarFp.getMilestones(status, limit, options).then((request) => request(axios, basePath));
        },
    };
};
exports.MilestonesApiFactory = MilestonesApiFactory;
/**
 * MilestonesApi - object-oriented interface
 */
class MilestonesApi extends base_1.BaseAPI {
    /**
     * Get a single milestone by ID
     * @summary Get Milestone
     * @param {string} milestoneId Milestone ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMilestone(milestoneId, options) {
        return (0, exports.MilestonesApiFp)(this.configuration).getMilestone(milestoneId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get all milestones
     * @summary Get Milestones
     * @param {string} [status] Filter by milestone status
     * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMilestones(status, limit, options) {
        return (0, exports.MilestonesApiFp)(this.configuration).getMilestones(status, limit, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.MilestonesApi = MilestonesApi;
