// kalshi-typescript/dist/esm/api/collection-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 2.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// URLSearchParams not necessarily used
// @ts-ignore
import { URL } from 'url';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setApiKeyToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * CollectionApi - axios parameter creator
 */
export const CollectionApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *  Endpoint for looking up an individual market in a multivariate event collection. This endpoint must be hit at least once before trading or looking up a market.
         * @summary Create Market In Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {CreateMarketInMultivariateEventCollectionRequest} createMarketInMultivariateEventCollectionRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createMarketInMultivariateEventCollection: (collectionTicker_1, createMarketInMultivariateEventCollectionRequest_1, ...args_1) => __awaiter(this, [collectionTicker_1, createMarketInMultivariateEventCollectionRequest_1, ...args_1], void 0, function* (collectionTicker, createMarketInMultivariateEventCollectionRequest, options = {}) {
            // verify required parameter 'collectionTicker' is not null or undefined
            assertParamExists('createMarketInMultivariateEventCollection', 'collectionTicker', collectionTicker);
            // verify required parameter 'createMarketInMultivariateEventCollectionRequest' is not null or undefined
            assertParamExists('createMarketInMultivariateEventCollection', 'createMarketInMultivariateEventCollectionRequest', createMarketInMultivariateEventCollectionRequest);
            const localVarPath = `/multivariate_event_collections/{collection_ticker}`
                .replace(`{${"collection_ticker"}}`, encodeURIComponent(String(collectionTicker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication kalshiAccessSignature required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-SIGNATURE", configuration);
            // authentication kalshiAccessKey required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-KEY", configuration);
            // authentication kalshiAccessTimestamp required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-TIMESTAMP", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = serializeDataIfNeeded(createMarketInMultivariateEventCollectionRequest, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *  Endpoint for getting data about a multivariate event collection by its ticker.
         * @summary Get Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollection: (collectionTicker_1, ...args_1) => __awaiter(this, [collectionTicker_1, ...args_1], void 0, function* (collectionTicker, options = {}) {
            // verify required parameter 'collectionTicker' is not null or undefined
            assertParamExists('getMultivariateEventCollection', 'collectionTicker', collectionTicker);
            const localVarPath = `/multivariate_event_collections/{collection_ticker}`
                .replace(`{${"collection_ticker"}}`, encodeURIComponent(String(collectionTicker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *  Endpoint for retrieving which markets in an event collection were recently looked up.
         * @summary Get Multivariate Event Collection Lookup History
         * @param {string} collectionTicker Collection ticker
         * @param {GetMultivariateEventCollectionLookupHistoryLookbackSecondsEnum} lookbackSeconds Number of seconds to look back for lookup history. Must be one of 10, 60, 300, or 3600.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollectionLookupHistory: (collectionTicker_1, lookbackSeconds_1, ...args_1) => __awaiter(this, [collectionTicker_1, lookbackSeconds_1, ...args_1], void 0, function* (collectionTicker, lookbackSeconds, options = {}) {
            // verify required parameter 'collectionTicker' is not null or undefined
            assertParamExists('getMultivariateEventCollectionLookupHistory', 'collectionTicker', collectionTicker);
            // verify required parameter 'lookbackSeconds' is not null or undefined
            assertParamExists('getMultivariateEventCollectionLookupHistory', 'lookbackSeconds', lookbackSeconds);
            const localVarPath = `/multivariate_event_collections/{collection_ticker}/lookup`
                .replace(`{${"collection_ticker"}}`, encodeURIComponent(String(collectionTicker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (lookbackSeconds !== undefined) {
                localVarQueryParameter['lookback_seconds'] = lookbackSeconds;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *  Endpoint for getting data about multivariate event collections.
         * @summary Get Multivariate Event Collections
         * @param {GetMultivariateEventCollectionsStatusEnum} [status] Only return collections of a certain status. Can be unopened, open, or closed.
         * @param {string} [associatedEventTicker] Only return collections associated with a particular event ticker.
         * @param {string} [seriesTicker] Only return collections with a particular series ticker.
         * @param {number} [limit] Specify the maximum number of results.
         * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. This optional parameter, when filled, should be filled with the cursor string returned in a previous request to this end-point.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollections: (status_1, associatedEventTicker_1, seriesTicker_1, limit_1, cursor_1, ...args_1) => __awaiter(this, [status_1, associatedEventTicker_1, seriesTicker_1, limit_1, cursor_1, ...args_1], void 0, function* (status, associatedEventTicker, seriesTicker, limit, cursor, options = {}) {
            const localVarPath = `/multivariate_event_collections`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (status !== undefined) {
                localVarQueryParameter['status'] = status;
            }
            if (associatedEventTicker !== undefined) {
                localVarQueryParameter['associated_event_ticker'] = associatedEventTicker;
            }
            if (seriesTicker !== undefined) {
                localVarQueryParameter['series_ticker'] = seriesTicker;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (cursor !== undefined) {
                localVarQueryParameter['cursor'] = cursor;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *  Endpoint for looking up an individual market in a multivariate event collection. If CreateMarketInMultivariateEventCollection has never been hit with that variable combination before, this will return a 404.
         * @summary Lookup Tickers For Market In Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {LookupTickersForMarketInMultivariateEventCollectionRequest} lookupTickersForMarketInMultivariateEventCollectionRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupTickersForMarketInMultivariateEventCollection: (collectionTicker_1, lookupTickersForMarketInMultivariateEventCollectionRequest_1, ...args_1) => __awaiter(this, [collectionTicker_1, lookupTickersForMarketInMultivariateEventCollectionRequest_1, ...args_1], void 0, function* (collectionTicker, lookupTickersForMarketInMultivariateEventCollectionRequest, options = {}) {
            // verify required parameter 'collectionTicker' is not null or undefined
            assertParamExists('lookupTickersForMarketInMultivariateEventCollection', 'collectionTicker', collectionTicker);
            // verify required parameter 'lookupTickersForMarketInMultivariateEventCollectionRequest' is not null or undefined
            assertParamExists('lookupTickersForMarketInMultivariateEventCollection', 'lookupTickersForMarketInMultivariateEventCollectionRequest', lookupTickersForMarketInMultivariateEventCollectionRequest);
            const localVarPath = `/multivariate_event_collections/{collection_ticker}/lookup`
                .replace(`{${"collection_ticker"}}`, encodeURIComponent(String(collectionTicker)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'PUT' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication kalshiAccessSignature required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-SIGNATURE", configuration);
            // authentication kalshiAccessKey required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-KEY", configuration);
            // authentication kalshiAccessTimestamp required
            yield setApiKeyToObject(localVarHeaderParameter, "KALSHI-ACCESS-TIMESTAMP", configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = serializeDataIfNeeded(lookupTickersForMarketInMultivariateEventCollectionRequest, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * CollectionApi - functional programming interface
 */
export const CollectionApiFp = function (configuration) {
    const localVarAxiosParamCreator = CollectionApiAxiosParamCreator(configuration);
    return {
        /**
         *  Endpoint for looking up an individual market in a multivariate event collection. This endpoint must be hit at least once before trading or looking up a market.
         * @summary Create Market In Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {CreateMarketInMultivariateEventCollectionRequest} createMarketInMultivariateEventCollectionRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createMarketInMultivariateEventCollection(collectionTicker, createMarketInMultivariateEventCollectionRequest, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createMarketInMultivariateEventCollection(collectionTicker, createMarketInMultivariateEventCollectionRequest, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         *  Endpoint for getting data about a multivariate event collection by its ticker.
         * @summary Get Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollection(collectionTicker, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMultivariateEventCollection(collectionTicker, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         *  Endpoint for retrieving which markets in an event collection were recently looked up.
         * @summary Get Multivariate Event Collection Lookup History
         * @param {string} collectionTicker Collection ticker
         * @param {GetMultivariateEventCollectionLookupHistoryLookbackSecondsEnum} lookbackSeconds Number of seconds to look back for lookup history. Must be one of 10, 60, 300, or 3600.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollectionLookupHistory(collectionTicker, lookbackSeconds, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMultivariateEventCollectionLookupHistory(collectionTicker, lookbackSeconds, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         *  Endpoint for getting data about multivariate event collections.
         * @summary Get Multivariate Event Collections
         * @param {GetMultivariateEventCollectionsStatusEnum} [status] Only return collections of a certain status. Can be unopened, open, or closed.
         * @param {string} [associatedEventTicker] Only return collections associated with a particular event ticker.
         * @param {string} [seriesTicker] Only return collections with a particular series ticker.
         * @param {number} [limit] Specify the maximum number of results.
         * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. This optional parameter, when filled, should be filled with the cursor string returned in a previous request to this end-point.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollections(status, associatedEventTicker, seriesTicker, limit, cursor, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMultivariateEventCollections(status, associatedEventTicker, seriesTicker, limit, cursor, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         *  Endpoint for looking up an individual market in a multivariate event collection. If CreateMarketInMultivariateEventCollection has never been hit with that variable combination before, this will return a 404.
         * @summary Lookup Tickers For Market In Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {LookupTickersForMarketInMultivariateEventCollectionRequest} lookupTickersForMarketInMultivariateEventCollectionRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupTickersForMarketInMultivariateEventCollection(collectionTicker, lookupTickersForMarketInMultivariateEventCollectionRequest, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.lookupTickersForMarketInMultivariateEventCollection(collectionTicker, lookupTickersForMarketInMultivariateEventCollectionRequest, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
/**
 * CollectionApi - factory interface
 */
export const CollectionApiFactory = function (configuration, basePath, axios) {
    const localVarFp = CollectionApiFp(configuration);
    return {
        /**
         *  Endpoint for looking up an individual market in a multivariate event collection. This endpoint must be hit at least once before trading or looking up a market.
         * @summary Create Market In Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {CreateMarketInMultivariateEventCollectionRequest} createMarketInMultivariateEventCollectionRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createMarketInMultivariateEventCollection(collectionTicker, createMarketInMultivariateEventCollectionRequest, options) {
            return localVarFp.createMarketInMultivariateEventCollection(collectionTicker, createMarketInMultivariateEventCollectionRequest, options).then((request) => request(axios, basePath));
        },
        /**
         *  Endpoint for getting data about a multivariate event collection by its ticker.
         * @summary Get Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollection(collectionTicker, options) {
            return localVarFp.getMultivariateEventCollection(collectionTicker, options).then((request) => request(axios, basePath));
        },
        /**
         *  Endpoint for retrieving which markets in an event collection were recently looked up.
         * @summary Get Multivariate Event Collection Lookup History
         * @param {string} collectionTicker Collection ticker
         * @param {GetMultivariateEventCollectionLookupHistoryLookbackSecondsEnum} lookbackSeconds Number of seconds to look back for lookup history. Must be one of 10, 60, 300, or 3600.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollectionLookupHistory(collectionTicker, lookbackSeconds, options) {
            return localVarFp.getMultivariateEventCollectionLookupHistory(collectionTicker, lookbackSeconds, options).then((request) => request(axios, basePath));
        },
        /**
         *  Endpoint for getting data about multivariate event collections.
         * @summary Get Multivariate Event Collections
         * @param {GetMultivariateEventCollectionsStatusEnum} [status] Only return collections of a certain status. Can be unopened, open, or closed.
         * @param {string} [associatedEventTicker] Only return collections associated with a particular event ticker.
         * @param {string} [seriesTicker] Only return collections with a particular series ticker.
         * @param {number} [limit] Specify the maximum number of results.
         * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. This optional parameter, when filled, should be filled with the cursor string returned in a previous request to this end-point.
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMultivariateEventCollections(status, associatedEventTicker, seriesTicker, limit, cursor, options) {
            return localVarFp.getMultivariateEventCollections(status, associatedEventTicker, seriesTicker, limit, cursor, options).then((request) => request(axios, basePath));
        },
        /**
         *  Endpoint for looking up an individual market in a multivariate event collection. If CreateMarketInMultivariateEventCollection has never been hit with that variable combination before, this will return a 404.
         * @summary Lookup Tickers For Market In Multivariate Event Collection
         * @param {string} collectionTicker Collection ticker
         * @param {LookupTickersForMarketInMultivariateEventCollectionRequest} lookupTickersForMarketInMultivariateEventCollectionRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        lookupTickersForMarketInMultivariateEventCollection(collectionTicker, lookupTickersForMarketInMultivariateEventCollectionRequest, options) {
            return localVarFp.lookupTickersForMarketInMultivariateEventCollection(collectionTicker, lookupTickersForMarketInMultivariateEventCollectionRequest, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * CollectionApi - object-oriented interface
 */
export class CollectionApi extends BaseAPI {
    /**
     *  Endpoint for looking up an individual market in a multivariate event collection. This endpoint must be hit at least once before trading or looking up a market.
     * @summary Create Market In Multivariate Event Collection
     * @param {string} collectionTicker Collection ticker
     * @param {CreateMarketInMultivariateEventCollectionRequest} createMarketInMultivariateEventCollectionRequest
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    createMarketInMultivariateEventCollection(collectionTicker, createMarketInMultivariateEventCollectionRequest, options) {
        return CollectionApiFp(this.configuration).createMarketInMultivariateEventCollection(collectionTicker, createMarketInMultivariateEventCollectionRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *  Endpoint for getting data about a multivariate event collection by its ticker.
     * @summary Get Multivariate Event Collection
     * @param {string} collectionTicker Collection ticker
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMultivariateEventCollection(collectionTicker, options) {
        return CollectionApiFp(this.configuration).getMultivariateEventCollection(collectionTicker, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *  Endpoint for retrieving which markets in an event collection were recently looked up.
     * @summary Get Multivariate Event Collection Lookup History
     * @param {string} collectionTicker Collection ticker
     * @param {GetMultivariateEventCollectionLookupHistoryLookbackSecondsEnum} lookbackSeconds Number of seconds to look back for lookup history. Must be one of 10, 60, 300, or 3600.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMultivariateEventCollectionLookupHistory(collectionTicker, lookbackSeconds, options) {
        return CollectionApiFp(this.configuration).getMultivariateEventCollectionLookupHistory(collectionTicker, lookbackSeconds, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *  Endpoint for getting data about multivariate event collections.
     * @summary Get Multivariate Event Collections
     * @param {GetMultivariateEventCollectionsStatusEnum} [status] Only return collections of a certain status. Can be unopened, open, or closed.
     * @param {string} [associatedEventTicker] Only return collections associated with a particular event ticker.
     * @param {string} [seriesTicker] Only return collections with a particular series ticker.
     * @param {number} [limit] Specify the maximum number of results.
     * @param {string} [cursor] The Cursor represents a pointer to the next page of records in the pagination. This optional parameter, when filled, should be filled with the cursor string returned in a previous request to this end-point.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMultivariateEventCollections(status, associatedEventTicker, seriesTicker, limit, cursor, options) {
        return CollectionApiFp(this.configuration).getMultivariateEventCollections(status, associatedEventTicker, seriesTicker, limit, cursor, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *  Endpoint for looking up an individual market in a multivariate event collection. If CreateMarketInMultivariateEventCollection has never been hit with that variable combination before, this will return a 404.
     * @summary Lookup Tickers For Market In Multivariate Event Collection
     * @param {string} collectionTicker Collection ticker
     * @param {LookupTickersForMarketInMultivariateEventCollectionRequest} lookupTickersForMarketInMultivariateEventCollectionRequest
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    lookupTickersForMarketInMultivariateEventCollection(collectionTicker, lookupTickersForMarketInMultivariateEventCollectionRequest, options) {
        return CollectionApiFp(this.configuration).lookupTickersForMarketInMultivariateEventCollection(collectionTicker, lookupTickersForMarketInMultivariateEventCollectionRequest, options).then((request) => request(this.axios, this.basePath));
    }
}
export const GetMultivariateEventCollectionLookupHistoryLookbackSecondsEnum = {
    NUMBER_10: 10,
    NUMBER_60: 60,
    NUMBER_300: 300,
    NUMBER_3600: 3600
};
export const GetMultivariateEventCollectionsStatusEnum = {
    Unopened: 'unopened',
    Open: 'open',
    Closed: 'closed'
};
