// kalshi-typescript/dist/esm/api/milestones-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trading API
 * Complete API for the Kalshi trading platform including all handlers for SDK generation
 *
 * The version of the OpenAPI document: 2.0.0
 * Contact: support@kalshi.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// URLSearchParams not necessarily used
// @ts-ignore
import { URL } from 'url';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setBearerAuthToObject, setSearchParams, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * MilestonesApi - axios parameter creator
 */
export const MilestonesApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Get a single milestone by ID
         * @summary Get Milestone
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestone: (milestoneId_1, ...args_1) => __awaiter(this, [milestoneId_1, ...args_1], void 0, function* (milestoneId, options = {}) {
            // verify required parameter 'milestoneId' is not null or undefined
            assertParamExists('getMilestone', 'milestoneId', milestoneId);
            const localVarPath = `/milestones/{milestone_id}`
                .replace(`{${"milestone_id"}}`, encodeURIComponent(String(milestoneId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Get all milestones
         * @summary Get Milestones
         * @param {string} [status] Filter by milestone status
         * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestones: (status_1, limit_1, ...args_1) => __awaiter(this, [status_1, limit_1, ...args_1], void 0, function* (status, limit, options = {}) {
            const localVarPath = `/milestones`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearerAuth required
            // http bearer authentication required
            yield setBearerAuthToObject(localVarHeaderParameter, configuration);
            if (status !== undefined) {
                localVarQueryParameter['status'] = status;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * MilestonesApi - functional programming interface
 */
export const MilestonesApiFp = function (configuration) {
    const localVarAxiosParamCreator = MilestonesApiAxiosParamCreator(configuration);
    return {
        /**
         * Get a single milestone by ID
         * @summary Get Milestone
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestone(milestoneId, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMilestone(milestoneId, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Get all milestones
         * @summary Get Milestones
         * @param {string} [status] Filter by milestone status
         * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestones(status, limit, options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getMilestones(status, limit, options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
/**
 * MilestonesApi - factory interface
 */
export const MilestonesApiFactory = function (configuration, basePath, axios) {
    const localVarFp = MilestonesApiFp(configuration);
    return {
        /**
         * Get a single milestone by ID
         * @summary Get Milestone
         * @param {string} milestoneId Milestone ID
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestone(milestoneId, options) {
            return localVarFp.getMilestone(milestoneId, options).then((request) => request(axios, basePath));
        },
        /**
         * Get all milestones
         * @summary Get Milestones
         * @param {string} [status] Filter by milestone status
         * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getMilestones(status, limit, options) {
            return localVarFp.getMilestones(status, limit, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * MilestonesApi - object-oriented interface
 */
export class MilestonesApi extends BaseAPI {
    /**
     * Get a single milestone by ID
     * @summary Get Milestone
     * @param {string} milestoneId Milestone ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMilestone(milestoneId, options) {
        return MilestonesApiFp(this.configuration).getMilestone(milestoneId, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Get all milestones
     * @summary Get Milestones
     * @param {string} [status] Filter by milestone status
     * @param {number} [limit] Number of items per page (minimum 1, maximum 500)
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getMilestones(status, limit, options) {
        return MilestonesApiFp(this.configuration).getMilestones(status, limit, options).then((request) => request(this.axios, this.basePath));
    }
}
