// kalshi-typescript/dist/esm/api/search-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// URLSearchParams not necessarily used
// @ts-ignore
import { URL } from 'url';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, setSearchParams, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * SearchApi - axios parameter creator
 */
export const SearchApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
         * @summary Get Filters for Sports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFiltersForSports: (...args_1) => __awaiter(this, [...args_1], void 0, function* (options = {}) {
            const localVarPath = `/search/filters_by_sport`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
         * @summary Get Tags for Series Categories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTagsForSeriesCategories: (...args_1) => __awaiter(this, [...args_1], void 0, function* (options = {}) {
            const localVarPath = `/search/tags_by_categories`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * SearchApi - functional programming interface
 */
export const SearchApiFp = function (configuration) {
    const localVarAxiosParamCreator = SearchApiAxiosParamCreator(configuration);
    return {
        /**
         * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
         * @summary Get Filters for Sports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFiltersForSports(options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getFiltersForSports(options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
        /**
         * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
         * @summary Get Tags for Series Categories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTagsForSeriesCategories(options) {
            return __awaiter(this, void 0, void 0, function* () {
                var _a;
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getTagsForSeriesCategories(options);
                const localVarOperationServerIndex = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const localVarOperationServerBasePath = undefined;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, localVarOperationServerBasePath || basePath);
            });
        },
    };
};
/**
 * SearchApi - factory interface
 */
export const SearchApiFactory = function (configuration, basePath, axios) {
    const localVarFp = SearchApiFp(configuration);
    return {
        /**
         * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
         * @summary Get Filters for Sports
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getFiltersForSports(options) {
            return localVarFp.getFiltersForSports(options).then((request) => request(axios, basePath));
        },
        /**
         * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
         * @summary Get Tags for Series Categories
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTagsForSeriesCategories(options) {
            return localVarFp.getTagsForSeriesCategories(options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * SearchApi - object-oriented interface
 */
export class SearchApi extends BaseAPI {
    /**
     * Retrieve available filters organized by sport.  This endpoint returns filtering options available for each sport, including scopes and competitions. It also provides an ordered list of sports for display purposes.
     * @summary Get Filters for Sports
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getFiltersForSports(options) {
        return SearchApiFp(this.configuration).getFiltersForSports(options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Retrieve tags organized by series categories.  This endpoint returns a mapping of series categories to their associated tags, which can be used for filtering and search functionality.
     * @summary Get Tags for Series Categories
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getTagsForSeriesCategories(options) {
        return SearchApiFp(this.configuration).getTagsForSeriesCategories(options).then((request) => request(this.axios, this.basePath));
    }
}
