// kalshi-typescript/dist/esm/models/market.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
export const MarketMarketTypeEnum = {
    Binary: 'binary',
    Scalar: 'scalar'
};
export const MarketStatusEnum = {
    Initialized: 'initialized',
    Inactive: 'inactive',
    Active: 'active',
    Closed: 'closed',
    Determined: 'determined',
    Disputed: 'disputed',
    Amended: 'amended',
    Finalized: 'finalized'
};
export const MarketResponsePriceUnitsEnum = {
    UsdCent: 'usd_cent'
};
export const MarketResultEnum = {
    Yes: 'yes',
    No: 'no',
    Empty: ''
};
export const MarketStrikeTypeEnum = {
    Greater: 'greater',
    GreaterOrEqual: 'greater_or_equal',
    Less: 'less',
    LessOrEqual: 'less_or_equal',
    Between: 'between',
    Functional: 'functional',
    Custom: 'custom',
    Structured: 'structured'
};
