// kalshi-typescript/dist/models/create-order-request.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrderRequestTimeInForceEnum = exports.CreateOrderRequestTypeEnum = exports.CreateOrderRequestActionEnum = exports.CreateOrderRequestSideEnum = void 0;
exports.CreateOrderRequestSideEnum = {
    Yes: 'yes',
    No: 'no'
};
exports.CreateOrderRequestActionEnum = {
    Buy: 'buy',
    Sell: 'sell'
};
exports.CreateOrderRequestTypeEnum = {
    Limit: 'limit',
    Market: 'market'
};
exports.CreateOrderRequestTimeInForceEnum = {
    FillOrKill: 'fill_or_kill',
    GoodTillCanceled: 'good_till_canceled',
    ImmediateOrCancel: 'immediate_or_cancel'
};
