// kalshi-typescript/dist/models/market.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kalshi Trade API Manual Endpoints
 * Manually defined OpenAPI spec for endpoints being migrated to spec-first approach
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MarketStrikeTypeEnum = exports.MarketResultEnum = exports.MarketResponsePriceUnitsEnum = exports.MarketStatusEnum = exports.MarketMarketTypeEnum = void 0;
exports.MarketMarketTypeEnum = {
    Binary: 'binary',
    Scalar: 'scalar'
};
exports.MarketStatusEnum = {
    Initialized: 'initialized',
    Inactive: 'inactive',
    Active: 'active',
    Closed: 'closed',
    Determined: 'determined',
    Disputed: 'disputed',
    Amended: 'amended',
    Finalized: 'finalized'
};
exports.MarketResponsePriceUnitsEnum = {
    UsdCent: 'usd_cent'
};
exports.MarketResultEnum = {
    Yes: 'yes',
    No: 'no',
    Empty: ''
};
exports.MarketStrikeTypeEnum = {
    Greater: 'greater',
    GreaterOrEqual: 'greater_or_equal',
    Less: 'less',
    LessOrEqual: 'less_or_equal',
    Between: 'between',
    Functional: 'functional',
    Custom: 'custom',
    Structured: 'structured'
};
